### PR TITLE
Add TSYS01 simulator

### PR DESCRIPTION
**Pull request by peterbarker**
Will help with https://github.com/ArduPilot/ardupilot/pull/14239

### CODE REVIEW

**Comment by peterbarker** [comment]
@patrickelectric @jaxxzer I'd like to merge this one.

Means that the `TSYS01` driver gets some testing in autotest.

**Review by tridge — APPROVED** [review]

File: libraries/SITL/SIM_Submarine.cpp [hunk 1]
```diff
@@ -235,6 +235,7 @@ void Submarine::update(const struct sitl_input &input)
     calculate_forces(input, rot_accel, accel_body);
 
     update_dynamics(rot_accel);
+    update_external_payload(input);
```
Comment (jaxxzer) [inline libraries/SITL/SIM_Submarine.cpp:238]: what does this do?

**Review by jaxxzer — APPROVED** [review]
nice thanks

File: libraries/SITL/SIM_Submarine.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/SITL/SIM_Submarine.cpp:238]: It adds hooks for ancillary things not directly related to the vehicle dynamics.

We'll probably want to move the simulated i2c device updates out of that at some stage, but that's where they're currently hooked.

**Pull request merged**